flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -1212,6 +1297,44 @@ void createTemporarySystemFunction(
      */
     boolean dropView(String path, boolean ignoreIfNotExists);
 
+    /**
+     * Drops a model registered in the given path.
+     *
+     * <p>This method can only drop permanent objects. Temporary objects can shadow permanent ones.
+     * If a temporary object exists in a given path, make sure to drop the temporary object first
+     * using {@link #dropTemporaryModel}.
+     *
+     * @param path The given path under which the model will be dropped. See also the {@link
+     *     TableEnvironment} class description for the format of the path.
+     */
+    boolean dropModel(String path);
+
+    /**
+     * Drops a model registered in the given path.
+     *
+     * <p>This method can only drop permanent objects. Temporary objects can shadow permanent ones.
+     * If a temporary object exists in a given path, make sure to drop the temporary object first
+     * using {@link #dropTemporaryModel}.
+     *
+     * @param path The given path under which the model will be dropped. See also the {@link
+     *     TableEnvironment} class description for the format of the path.
+     * @param ignoreIfNotExists If false exception will be thrown if the model to drop does not
+     *     exist.
+     */
+    boolean dropModel(String path, boolean ignoreIfNotExists);
+
+    /**
+     * Drops a temporary model registered in the given path.
+     *
+     * <p>If a permanent model with a given path exists, it will be used from now on for any queries
+     * that reference this path.
+     *
+     * @param path The given path under which the temporary model will be dropped. See also the
+     *     {@link TableEnvironment} class description for the format of the path.
+     * @return true if a model existed in the given path and was removed
+     */
+    boolean dropTemporaryModel(String path);
```

Comment:
can we have junits , of this method , with a null path, valid path invalid path and with the flag set and not set please.